python/pyspark/sql/functions.py

```diff
@@ -2129,8 +2130,12 @@ def _create_udf(f, returnType, vectorized):
     def _udf(f, returnType=StringType(), vectorized=vectorized):
         if vectorized:
             import inspect
-            if len(inspect.getargspec(f).args) == 0:
-                raise NotImplementedError("0-parameter pandas_udfs are not currently supported")
+            argspec = inspect.getargspec(f)
+            if len(argspec.args) == 0 and argspec.varargs is None:
```

Comment:
I hadn't really thought about it, but does this mean varargs are supported?  I suppose it could, but maybe best to include a test for it.

Reply:
I think `varargs` are fine. I will add the test.

Reply:
Added.